src/gt4sd/algorithms/generation/diffusion/core.py

```diff
@@ -0,0 +1,353 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+"""HuggingFace Diffusers generation algorithm. Code and models adapted from https://github.com/huggingface/diffusers."""
+
+import logging
+from dataclasses import field
+from typing import ClassVar, Dict, Optional, Set, TypeVar
+
+from ...core import (
+    AlgorithmConfiguration,
+    GeneratorAlgorithm,
+    Untargeted,
+    get_configuration_class_with_attributes,
+)
+from ...registry import ApplicationsRegistry
+from .implementation import MODEL_TYPES, SCHEDULER_TYPES, Generator
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+T = type(None)
+S = TypeVar("S", bound=str)
+
+
+class DiffusersGenerationAlgorithm(GeneratorAlgorithm[S, T]):
+    def __init__(
+        self, configuration: AlgorithmConfiguration, target: Optional[T] = None
+    ):
+        """Diffusers generation algorithm.
+
+        Args:
+            configuration: domain and application
+                specification, defining types and validations.
+            target: unused since it is not a conditional generator.
+
+        Example:
+            An example for using a generative algorithm from Diffusers::
+
+                configuration = GeneratorConfiguration()
+                algorithm = DiffusersGenerationAlgorithm(configuration=configuration)
+                items = list(algorithm.sample(1))
+                print(items)
+        """
+
+        configuration = self.validate_configuration(configuration)
+        # TODO there might also be a validation/check on the target input
+
+        super().__init__(
+            configuration=configuration,
+            target=None,  # type:ignore
+        )
+
+    def get_generator(
+        self,
+        configuration: AlgorithmConfiguration[S, T],
+        target: Optional[T],
+    ) -> Untargeted:
+        """Get the function to sample batches.
+
+        Args:
+            configuration: helps to set up the application.
+            target: context or condition for the generation. Unused in the algorithm.
+
+        Returns:
+            callable generating a batch of items.
+        """
+        logger.info("ensure artifacts for the application are present.")
+        self.local_artifacts = configuration.ensure_artifacts()
+        implementation: Generator = configuration.get_conditional_generator(  # type: ignore
+            self.local_artifacts
+        )
+        return implementation.sample
+
+    def validate_configuration(
+        self, configuration: AlgorithmConfiguration
+    ) -> AlgorithmConfiguration:
+        assert isinstance(configuration, AlgorithmConfiguration)
+        return configuration
+
+
+@ApplicationsRegistry.register_algorithm_application(DiffusersGenerationAlgorithm)
+class DiffusersConfiguration(AlgorithmConfiguration[str, None]):
```

Comment:
I have the feeling some parameters are coming right away from the HuggingFace wrappers. A lot of them are unused in the implementation.py, e.g., repetition_penalty.
Let's try to clean those up and also to capture what we pass to the sample method and parameters of the configuration.